src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs

```diff
@@ -246,36 +247,32 @@ private static string GetIndentedXmlString(XmlDocument doc)
         /// returns a path to the project file which defines the benchmarks
         /// </summary>
         [PublicAPI]
-        protected virtual FileInfo GetProjectFilePath(Type benchmarkTarget, ILogger logger)
+        protected virtual FileInfo GetProjectFilePath(BenchmarkCase benchmark, ILogger logger)
         {
+            var benchmarkTarget = benchmark.Descriptor.Type;
+
             if (!GetSolutionRootDirectory(out var rootDirectory) && !GetProjectRootDirectory(out rootDirectory))
             {
                 logger.WriteLineError(
                     $"Unable to find .sln or .csproj file. Will use current directory {Directory.GetCurrentDirectory()} to search for project file. If you don't use .sln file on purpose it should not be a problem.");
                 rootDirectory = new DirectoryInfo(Directory.GetCurrentDirectory());
             }
 
-            // important assumption! project's file name === output dll name
-            string projectName = benchmarkTarget.GetTypeInfo().Assembly.GetName().Name;
+            List<string> notFound = new List<string>();
+            foreach (ILocator locator in benchmark.Config.GetLocators())
```

Comment:
I think the default locator should be filtered out, and used last if none of the custom locators return a valid path.

Reply:
It might make more sense to let the default locator stay in `GetProjectFilePath` and run the locators first then. What do you think?

Reply:
Done